### PR TITLE
ENH: Transition project configuration to `pyproject.toml`

### PR DESCRIPTION
**Pull request by jhlegarreta**
- ENH: Transition project configuration to `pyproject.toml`
- COMP: Update CI configuration for ITK 5.4

### CODE REVIEW

**Review by aylward — APPROVED** [review]
LGTM!

If there are build failures that are unrelated, this can be merged.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+[build-system]
+requires = ["scikit-build-core"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "itk-minimalpathextraction"
+version = "1.0.0"
+description = "A minimal path extraction framework based on Fast Marching arrival functions."
+readme = "README.md"
```
Comment (thewtex) [inline pyproject.toml:9]: > pyproject_metadata.ConfigurationError: Readme file not found ("README.md")

-> README.rst

**Review by thewtex — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (jhlegarreta) [reply pyproject.toml]: :+1:. Changed in a9b5513.

**Review by thewtex — APPROVED** [review]
:sparkler:

**Pull request merged**